packages/cadl-ranch-specs/http/type/property/additional-properties/main.tsp

```diff
@@ -231,35 +256,306 @@ model ModelForRecord {
 
 #suppress "@azure-tools/typespec-azure-core/bad-record-type" "For testing"
 @doc("The model extends from Record<ModelForRecord> type.")
-model ExtendsModelAdditionalProperties extends Record<ModelForRecord> {}
+model ExtendsModelAdditionalProperties extends Record<ModelForRecord> {
+  knownProp: ModelForRecord;
+}
```

Comment:
why do we add the property?

Reply:
because previous extends and is Record way only works when the known property type is the same as the additional properties type.